manifests/fpm/config.pp

```diff
@@ -111,7 +111,15 @@
     mode    => '0644',
   }
 
-  ensure_resource('file', ['/var/run/php-fpm/', '/var/log/php-fpm/'], {
+  if (dirname($pid_file) == '/var/run/php-fpm') {
+    ensure_resource('file', ['/var/run/php-fpm/'], {
```

Comment:
Would it make sense to always create the `dirname($pid_file)` directory, and not only if it's `/var/run/php-fpm`?

Reply:
Well i see multiply issues if we always do that:
If $pid_file is something like /var/run/php-fpm.pid we would manage /var/run which is most likly something we don‘t want to do.

If $pid_file something complete custom (like /var/run/custom/location/php-fpm.pid) it would be kind of weird that we manage a part of it.

Best solution would be probably get rid of this totally and rely on the package management of the distribution to create the folders and adjust our defaults to represent the default configuration of the distribution.

Reply:
Honestly, I was surprised to discover the file resource in the code. I also had the assumption that systemd would create the directories during the service startup. I'm not sure what is the best way to proceed here.

Reply:
I removed it now entirly so that we rely on packet managment to create the default folders.
People which use custom locations should anyway care about the creation themself.

Tested already debian stretch and buster, which work perfectly with this change.